packages/fxa-auth-server/lib/metrics/glean/index.ts

```diff
@@ -109,13 +110,54 @@ const createEventFn =
         utm_term: metricsContext.utmTerm || '',
       };
 
+      // TODO: which events should have a reason?
+      const eventReason = metricsData?.reason || '';
+
       // uid needs extra handling because we need to hash the value
       const uid = findUid(request, metricsData);
       if (uid !== '') {
-        metrics.account_user_id_sha256 = sha256HashUid(uid);
+        commonMetrics.account_user_id_sha256 = sha256HashUid(uid);
       }
 
-      await gleanEventLogger.record(metrics);
+      // new style Glean events with event metric type
+      switch (eventName) {
```

Comment:
This works for me.  But if it bugs you to have this big switch statement in the middle here, we can define an object/map/hash where the key is the `eventName` and the value is the function, e.g.

```
const eventLoggers = {
  reg_acc_created: (x) =>
    gleanServerEventLogger.recordRegAccCreated(x),

  reg_acc_verified: ...
};
```

then call with `if (eventName in eventLoggers) { eventLoggers[eventName]({...commonMetrics, reason: eventReason, }); }`